tls/cfsslscan_handshake.go

```diff
@@ -22,9 +71,51 @@ func (c *Conn) SayHello() (cipherID, version uint16, err error) {
 	if err != nil {
 		return
 	}
+ 	if IsECcipherSuite(serverHello.cipherSuite) {
+		// Prime the connection, if necessary, for key
+		// exchange messages by reading off the certificate
+		// message and, if necessary, the OCSP stapling
+		// message
+		var msg interface{}
+		msg, err = c.readHandshake()
+		if err != nil {
+			return
+		}
+		certMsg, ok := msg.(*certificateMsg)
+		if !ok || len(certMsg.certificates) == 0 {
+			c.sendAlert(alertUnexpectedMessage)
+			err = unexpectedMessageError(certMsg, msg)
+			return
+		}
+		if serverHello.ocspStapling {
+			msg, err = c.readHandshake()
+			if err != nil {
+				return
+			}
+			_, ok := msg.(*certificateStatusMsg)
+			if !ok {
+				c.sendAlert(alertUnexpectedMessage)
+				return
+			}
+		}
+		
+		var skx *serverKeyExchangeMsg
+ 		skx, err = c.exchangeKeys()
+		if (skx.raw[0] != typeServerKeyExchange) {
+			panic("Bad handshake! Did not get ServerKeyExchange when expected!")
+		}
+ 		if err != nil {
```

Comment:
Check error conditions before operating on data. `err != nil` could imply `skx.raw` is empty.